### PR TITLE
port pitbulls from legacy

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
mostly the same as they were with some key changes:
- vent pitbulls have their own event instead of being tacked onto mouse migration
- if you fail to pet a pitbull it just retaliates instead of permanently making you KOS, meaning youd have to always kill the pitbull once someone failed to pet it. now they can run away and they will forget
- internally named Pitbull not Pibble
- made walter the base for pitbull to reduce copy paste and removing 90% of duplicate components from walter in an upstream file
- cargo order uses pitbull sprite instead of walter, false advertisement

## Why / Balance
poor old baby wouldnt hurt a fly
baby:

## Technical details
add `HostileOnInteractFail` that uses https://github.com/space-wizards/space-station-14/pull/32216

## Media
:trollface:

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ ] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
:cl:
- add: Pitbulls are back, couriers beware of what comes from the vents....

### CODE REVIEW

File: Resources/Prototypes/Nyanotrasen/Datasets/Names/pitbull.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - Cuddles
+  - Killer
+  - Paisley
+  - Echo
+  - Sweetie
+  - Tutu
+  - Reaper
+  - Peaches
+  - Cujo
+  - Cuddles
```
Comment (ghost) [inline Resources/Prototypes/Nyanotrasen/Datasets/Names/pitbull.yml]: Cuddles is in here twice, at line 16 and 25, is this intentional? Do we want to diversify the names of these?

**Review by ghost — CHANGES_REQUESTED** [review]
Looked at the current yaml, seems fine, but I noticed one thing. Also I can see TJ having opinions on this

File: Resources/Prototypes/Nyanotrasen/Datasets/Names/pitbull.yml (hunk 1 above)
Comment (deltanedas) [reply Resources/Prototypes/Nyanotrasen/Datasets/Names/pitbull.yml]: im gonna sort it too

**Comment by ghost** [comment]
pro

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Still needs Game Director and general maint approval

**Comment by deltanedas** [comment]
and needs upstream pr :trollface:

**Comment by VileSloth** [comment]
If these are going to be added, I suggest that they do not attack anyone for no reason, IE attacking a courier just because they are wearing courier clothes, as it perpetuates a stereotype that leads to animal abuse.

**Comment by TadJohnson00** [comment]
Hey, so, direction here.
We'd appreciate it if this wasn't brought back from Nyano. Sorry for the inconvenience

**Comment by deltanedas** [comment]
couriers on top

**Pull request closed without merging**